src/test/java/br/unicamp/cst/util/TestMemoryObserver.java

```diff
@@ -0,0 +1,261 @@
+package br.unicamp.cst.util;
+
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+
+import org.junit.Test;
+
+import br.unicamp.cst.core.entities.Codelet;
+import br.unicamp.cst.core.entities.Memory;
+import br.unicamp.cst.core.entities.MemoryContainer;
+import br.unicamp.cst.core.entities.MemoryObject;
+import br.unicamp.cst.core.entities.Mind;
+
+class CodeletToTest extends Codelet {
+
+	private int counter = 0;
+
+	public int getCounter() {
+		return counter;
+	}
+
+	public CodeletToTest(String name) {
+		setName(name);
+	}
+
+	@Override
+	public void accessMemoryObjects() {
+
+	}
+
+	@Override
+	public void calculateActivation() {
+
+	}
+
+	@Override
+	public void proc() {
+		counter++;
+	}
+
+}
+
+public class TestMemoryObserver {
+
+	@Test
+	public void test1() throws InterruptedException {
```

Comment:
please avoid generic names on tests as we would avoid it in other methods

```suggestion
	public void noMemoryChangeTest() throws InterruptedException {
```